gokey.go

```diff
@@ -58,6 +62,60 @@ func GetRaw(password, realm string, seed []byte, allowUnsafe bool) (io.Reader, e
 	return rng, nil
 }
 
+// below code implements asn1 encoding of x25519 and ed25519 keys according
+// to https://tools.ietf.org/id/draft-ietf-curdle-pkix-10.txt
+// the output should be compatible to OpenSSL pkey functions
+// this code is considered temporal and is expected to go away, when Go
+// implements native marshalling for these types of keys
+// as https://tools.ietf.org/id/draft-ietf-curdle-pkix-10.txt is still a draft
+// future versions of gokey may produce different output
+
+// p.3 https://tools.ietf.org/id/draft-ietf-curdle-pkix-10.txt
+// id-X25519    OBJECT IDENTIFIER ::= { 1 3 101 110 }
+// id-Ed25519   OBJECT IDENTIFIER ::= { 1 3 101 112 }
+const (
+	x25519OidSuffix  = 110
+	ed25519OidSuffix = 112
+)
+
+// x25519/ed25519 asn1 private key structure
+// p.7 https://tools.ietf.org/id/draft-ietf-curdle-pkix-10.txt
+// this implementation does not support optional attributes or public key
+type asn25519 struct {
+	Version    int
+	AlgId      pkix.AlgorithmIdentifier
+	PrivateKey []byte
+}
+
+// Golang does not have a declaration for x25519 keys
+type x25519PrivateKey []byte
+
+func marshal25519PrivateKey(key crypto.PrivateKey) ([]byte, error) {
+	var a25519 asn25519
+	a25519.Version = 0
```

Comment:
I think that's not needed. Go initializes to 0/nil/false by default

Reply:
Not sure if it is better to be explicit here, but probably you're right - will remove